saos-webapp/src/main/webapp/WEB-INF/static/javascript/analysis.js

```diff
@@ -1,80 +1,217 @@
+        
+        submitAndPrintAnalysisForm();
+    }
+    
+    
+    /****************** COMMON **/
+    
+    /**
+     * Submit (ajaxly) the analysisForm and reprints (and initiaties) it with the version (html) received 
+     * from the server endpoint.
+     *  
+     */
+    function submitAndPrintAnalysisForm() {
         
-        $('#analysisForm').append($('<input>').attr('id', actionParamName).attr('type', 'hidden').attr('name', actionParamName).val(actionParamValue));
         $('#analysisForm').attr('method', 'post');
         
         $('#analysisForm').ajaxSubmit(function(view) {
-            $('#analysisSearchCriteriaDiv').html(view);  
+            
+            $('#analysisFormDiv').html(view);  
+        
+            initButtons();
+            generateCharts();
+            
+        });
+    }
+    
+    
+    
+    
+    <!--$('input[id="daterange"]').daterangepicker();-->
+    
+    
+    
+    //******************************* MAIN CHART ***************************************************
+    
+    /**
+     * If the analysis form is not empty (uses #isAnalysisFormEmpty()),
+     * submits the analysisForm to a proper server endpoint and prints relevant charts based on
+     * data received from the server.
+     */
+    function generateCharts() {
         
-            $('#'+actionParamName).remove();
-            $('#analysisForm').attr('method', 'get');
+        if (!isAnalysisFormEmpty()) {
             
-            initButtonActions();
+            var analysisForm = $('#analysisForm');
+            var oldAction = analysisForm.attr('action');
+            analysisForm.attr('action', oldAction + "/generate");
             
+            analysisForm.ajaxSubmit(function(chart) {
+                 analysisForm.attr('action', oldAction)
+                 mainChart = chart;
+                 printMainChart(chart, null, null);
+             });
+         }
+
+    }
+    
+    /**
+     * Returns true if the analysis form fields inputs are empty
+     */
+    function isAnalysisFormEmpty() {
+        var empty = true;
+        $("#analysisForm :text").each(function() {
+            if($(this).val() !== "") {
+                empty = false;
+                return;
+            }
         });
+        
+        return empty;
+    }
+    
+    /**
+     * Prints main chart (number of judgments over the judgment date)
+     * @param chart json chart data, see pl.edu.icm.saos.webapp.chart.Chart 
+     * @param xmin min x value that will be shown on the chart, null - the min x value from chart data
+     * @param xmax max x value that will be shown on the chart, null - the max x value from chart data
+     */
+    function printMainChart(chart, xmin, xmax) {
+
+        var yMinMax = calculateYAxisRangeForXRange(chart, xmin, xmax);
+        var ymin = yMinMax[0];
+        var ymax = yMinMax[1];
+        
+        var seriesArr = [];
+        
+        for (var i = 0; i < chart.seriesList.length; i++) {
```

Comment:
Performance wise it is better to create variable for storing array length, than checking it every iteration. In this case you can write:
`for(var i = 0, length = chart.seriesList.length; i < length; i++) {`